src/library/basesqltablemodel.cpp

```diff
@@ -808,13 +808,20 @@ Qt::ItemFlags BaseSqlTableModel::readOnlyFlags(const QModelIndex &index) const {
     return defaultFlags;
 }
 
-const QLinkedList<int> BaseSqlTableModel::getTrackRows(TrackId trackId) const {
-    QHash<TrackId, QLinkedList<int> >::const_iterator it =
+QVector<int> BaseSqlTableModel::getTrackRows(TrackId trackId) const {
```

Comment:
The most efficient way will be returning the iterator it selves.
I think in all cases, the returned vector is used in a loop, so it should be possible.